src/App.tsx

```diff
@@ -1,27 +1,65 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [goodsRequest, setGoodsRequest] = useState('');
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  useEffect(() => {
+    switch (goodsRequest) {
+      case 'all':
+        getAll()
+          .then(setGoods);
+        break;
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+      case 'first 5':
+        get5First()
+          .then(setGoods);
+        break;
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+      case 'red':
+        getRedGoods()
+          .then(setGoods);
+        break;
 
-    <GoodsList goods={[]} />
-  </div>
-);
+      default:
+        break;
+    }
+  }, [goodsRequest]);
+
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
+
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={() => setGoodsRequest('all')}
```

Comment:
I'm not sure that it is a good idea:
You are setting "goodsRequest", your page rendering, and then getting data from the server, set it and make a second render with the data. So you have 2 renders, and get data only after the second render.

Will be better if you will call the method onClick which gets goods and set it immediately, so you will get all the data only in one render.

Reply:
To be honest, I don't understand what you mean.